### PR TITLE
Tested for posting a comment and related errors

### PR DESCRIPTION
**Pull request by HarryW217**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -121,6 +121,55 @@ describe("GET /api/articles", () => {
...
+
+        expect(typeof commentObj).toBe("object");
+        expect(typeof commentObj.comment_id).toBe("number");
+        expect(typeof commentObj.body).toBe("string");
+        expect(typeof commentObj.author).toBe("string");
+        expect(typeof commentObj.votes).toBe("number");
+        expect(typeof commentObj.created_at).toBe("string");
+      });
+  });
+  test("POST:400 responds with error message when body is missing required fields", () => {
```
Comment (Beth-McCormack) [inline __tests__/app.test.js:149]: Great tests to add👏

File: __tests__/app.test.js [hunk 2]
```diff
@@ -121,6 +121,55 @@ describe("GET /api/articles", () => {
...
+      body: 12345,
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(newComment)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid input");
+      });
+  });
+});
```
Comment (Beth-McCormack) [inline __tests__/app.test.js:197]: A couple more test cases to consider
- What if the article id doesnt exist eg api/articles/999/comments
- The article id is invalid eg api/articles/banana/comments

**Review by Beth-McCormack — COMMENTED** [review]
Nice work! I've left a few comments.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -168,6 +168,32 @@ describe("POST /api/articles/:article_id/comments", () => {
...
   });
+  test("POST:400 responds with error message when article id does not exist", () => {
+     const newComment = {
+       username: "icellusedkars",
+       body: "Hello world",
+     };
+     return request(app)
+       .post("/api/articles/99999/comments")
+       .send(newComment)
+       .expect(400)
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:179]: This would be a `404` - Does not exist

File: __tests__/app.test.js [hunk 4]
```diff
@@ -168,6 +168,32 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+  })
+  test("POST:400 responds with error message when article id is invalid", () => {
+    const newComment = {
+      username: "icellusedkars",
+      body: "Hello world",
+    };
+    return request(app)
+      .post("/api/articles/banana/comments")
+      .send(newComment)
+      .expect(400)
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:192]: This is correct , `400` - invalid input value

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Looks good just one of the `400` wants to be a `404` 👍 
No need to resubmit just make change and merge

**Pull request merged**